decode.go

```diff
@@ -70,7 +70,7 @@ func (de *decoder) message(buf []byte, sval reflect.Value) error {
 	// Decode all the fields
 	fields := ProtoFields(sval.Type())
 	fieldi := 0
-	for len(buf) > 0 {
+	for len(buf) > 0 && fieldi < len(fields) {
```

Comment:
Yes, let's go back to that one. Sorry, I was over-zealously trying protobuf to return useful errors for wrong inputs. But that's probably out of scope. Sorry for the next guy who gives too many bytes to protobuf...